### PR TITLE
add support for datadog monitors of type event-v2

### LINKED ISSUE #4456: datadog_monitor support for event-v2 monitors
### Summary

Datadog is deprecating the v1 event monitors in favor of a new v2 monitor type as documented https://docs.datadoghq.com/events/guides/migrating_to_new_events_features/ with the date of 6/30/2022 as the end date of v1 monitors.

They have been sending us emails about this, and I thought it was just a query change but it turns out that we need to set the type passed to the api to `event-v2 alert` instead of `event alert`

I have verified that this support can be added by adding the new event type into [datadog_monitor.py](https://github.com/ansible-collections/community.general/blob/main/plugins/modules/monitoring/datadog/datadog_monitor.py) and have the changes pushed to our fork and will PR from there

### Issue Type

Bug Report

### Component Name

community.general.datadog_monitor

### Ansible Version

```ansible --version
ansible [core 2.12.0]
  config file = None
  configured module search path = ['/Users/emeaney/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/emeaney/.pyenv/versions/3.10.0/envs/ansible3/lib/python3.10/site-packages/ansible
  ansible collection location = /Users/emeaney/.ansible/collections:/usr/share/ansible/collections
  executable location = /Users/emeaney/.pyenv/versions/ansible3/bin/ansible
  python version = 3.10.0 (default, Nov 30 2021, 15:16:48) [Clang 13.0.0 (clang-1300.0.29.3)]
  jinja version = 3.0.3
  libyaml = True

```

### Community.general Version

```console (paste below)
 ansible-galaxy collection list community.general

# /Users/emeaney/.pyenv/versions/3.10.0/envs/ansible3/lib/python3.10/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 4.0.2

```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

_No response_

### Steps to Reproduce

```yaml (paste below)
- community.general.datadog_monitor:
    name: !unsafe "[kubernetes] {{env.name}} Insufficient CPU messages are climbing"
    type: event-v2 alert
    state: present
    notification_message: !unsafe ""
    query: !unsafe "events(\"\\\"Insufficient cpu\\\"\").rollup(\"count\").by(\"env\").last(\"10m\") > 15"
    tags: ["integration:kubernetes"]
    notify_audit: false
    notify_no_data: false
    escalation_message: !unsafe ""
    renotify_interval: 10
    thresholds:
      critical: 15
      warning: 10
    api_key: "{{ dd_api_key }}"
    app_key: "{{ dd_app_key }}"
```

### Expected Results

A monitor of `event-v2 alert` is created in datadog

### Actual Results

```console (paste below)
you get an error about invalid type
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by ermeaney**
See https://docs.datadoghq.com/events/guides/migrating_to_new_events_features/

##### SUMMARY
Fixes #4456 

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

community.general.datadog_monitor
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

**Comment by ermeaney** [comment]
@felixfontein just updated the PR with the changelog fragment

**Comment by felixfontein** [comment]
No worries, I hid the messages :)

File: changelogs/fragments/4457-support-datadog-monitors-type event-v2.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
+  - datadog_monitor - support new datadog event monitor of type `event-v2 alert` (https://github.com/ansible-collections/community.general/pull/4457)
```
Comment (felixfontein) [inline changelogs/fragments/4457-support-datadog-monitors-type event-v2.yaml:3]:
```suggestion
  - datadog_monitor - support new datadog event monitor of type ``event-v2 alert`` (https://github.com/ansible-collections/community.general/pull/4457).
```

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
This looks good to me, in the sense that this will definitely not break anything that's currently working, so I don't mind merging it when nobody complains :) (I don't use datadog or the datadog modules, so that's all I can do ;) )

File: changelogs/fragments/4457-support-datadog-monitors-type event-v2.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+---
+bugfixes:
```
Comment (felixfontein) [inline changelogs/fragments/4457-support-datadog-monitors-type event-v2.yaml:2]: Ah, one thing: this is more a new feature than a bugfix, so it should be `minor_changes`.
```suggestion
minor_changes:
```

File: plugins/modules/monitoring/datadog/datadog_monitor.py [hunk 3]
```diff
@@ -56,6 +56,7 @@
             - metric alert
             - service check
             - event alert
+            - event-v2 alert
```
Comment (felixfontein) [inline plugins/modules/monitoring/datadog/datadog_monitor.py:59]: You also want to add
```
          - The type C(event-v2) was added in community.general 4.8.0.
```
in the `description` above.

File: changelogs/fragments/4457-support-datadog-monitors-type event-v2.yaml (hunk 2 above)
Comment (ermeaney) [reply changelogs/fragments/4457-support-datadog-monitors-type event-v2.yaml]: updated

File: plugins/modules/monitoring/datadog/datadog_monitor.py (hunk 3 above)
Comment (ermeaney) [reply plugins/modules/monitoring/datadog/datadog_monitor.py:60]: done

**Review by felixfontein — APPROVED** [review]

**Comment by felixfontein** [comment]
@ermeaney thanks for contributing this!

**Pull request merged**